testing/test_args.py

```diff
@@ -0,0 +1,36 @@
+import contextlib
+
+import pytest
+from selenium.webdriver.chrome.options import Options as ChromeOptions
+from webdriver_kaifuku import BrowserManager
+
+
+@pytest.mark.parametrize(
+    "conf,name",
+    [
+        (
+            {
+                "webdriver": "Remote",
+                "webdriver_options": {
+                    "desired_capabilities": {
+                        "browserName": "firefox",
+                    },
+                },
+            },
+            "firefox",
+        ),
+        (
+            {
+                "webdriver": "Remote",
+                "webdriver_options": {"options": ChromeOptions()},
+            },
+            "chrome",
+        ),
+    ],
+)
+def test_name(conf: dict, name: str):
+    """Browser name is read correctly"""
+    mgr = BrowserManager.from_conf(conf)
+    with contextlib.closing(mgr) as mgr:
```

Comment:
Maybe the discussion about the tests should wait until we have a decision on whether this change is worth and if so how it is implemented. However, before I forget I would like to leave it written here.
I was trying to find a better way to test this feature. I do not want to test `_get_browser_name` directly as that is probably going too deep into implementation details rather than focusing on expected behavior. However, I think for this to be testable we need a real browser so I am wondering if there is a way to monkeypatch the `create` and `close` method so that the browser connection is not made, only the WebDriver object is instantiated. Would that be possible? Would it make sense to try to do something like that?

Reply:
Is there a way to create them like that?

Reply:
I have been checking, and it might not be feasible, as capabilities are processed in the same [method] (https://github.com/SeleniumHQ/selenium/blob/036ee7f6aec797f5fe4608d2f7673e59d7f548b5/py/selenium/webdriver/remote/webdriver.py#L369) that executed the initialization.

The limitation then could be that we need to use remote driver for testing, but I think that is not a big limitation. Or do you have also infrastructure to test also non-remote drivers?